code/modules/admin/admin_ranks.dm

```diff
@@ -115,51 +118,101 @@ var/list/admin_ranks = list()								//list of all ranks with associated rights
 			load_admins()
 			return
 
-		var/DBQuery/query = dbcon.NewQuery("SELECT ckey, rank, (flags & 0xFFFF) as flags FROM ss13_player WHERE rank IS NOT NULL AND (flags & 0xFFFF) != 0")
+		var/DBQuery/query = dbcon.NewQuery("SELECT ckey, rank, flags FROM ss13_admins;")
 		query.Execute()
 		while(query.NextRow())
 			var/ckey = query.item[1]
 			var/rank = query.item[2]
 			var/rights = query.item[3]
-			if(istext(rights))	rights = text2num(rights)
+			if(istext(rights))
+				rights = text2num(rights)
+			
 			var/datum/admins/D = new /datum/admins(rank, rights, ckey)
-
 			//find the client for a ckey if they are connected and associate them with the new admin datum
 			D.associate(directory[ckey])
+
 		if(!admin_datums)
 			error("The database query in load_admins() resulted in no admins being added to the list. Reverting to legacy system.")
 			log_misc("The database query in load_admins() resulted in no admins being added to the list. Reverting to legacy system.")
 			config.admin_legacy_system = 1
 			load_admins()
 			return
 
-	#ifdef TESTING
+#ifdef TESTING
 	var/msg = "Admins Built:\n"
 	for(var/ckey in admin_datums)
 		var/rank
 		var/datum/admins/D = admin_datums[ckey]
 		if(D)	rank = D.rank
 		msg += "\t[ckey] - [rank]\n"
 	testing(msg)
-	#endif
+#endif
+
+/proc/clear_admins()
+	//clear the datums references
+	admin_datums.Cut()
+	for(var/s in staff)
+		var/client/C = s
+		C.remove_admin_verbs()
+		C.holder = null
+	staff.Cut()
 
+	// Clears admins from the world config.
+	for (var/A in world.GetConfig("admin"))
+		world.SetConfig("APP/admin", A, null)
 
-#ifdef TESTING
-/client/verb/changerank(newrank in admin_ranks)
-	if(holder)
-		holder.rank = newrank
-		holder.rights = admin_ranks[newrank]
-	else
-		holder = new /datum/admins(newrank,admin_ranks[newrank],ckey)
-	remove_admin_verbs()
-	holder.associate(src)
+/proc/update_admins_from_api(reload_once_done=FALSE)
+	set background = TRUE
 
-/client/verb/changerights(newrights as num)
-	if(holder)
-		holder.rights = newrights
-	else
-		holder = new /datum/admins("testing",newrights,ckey)
-	remove_admin_verbs()
-	holder.associate(src)
+	if (!establish_db_connection(dbcon))
+		return
 
-#endif
+	var/list/admins_to_push = list()
+
+	for (var/rank_name in admin_ranks)
+		var/datum/admin_rank/rank = admin_ranks[rank_name]
+		var/datum/http_request/forumuser_api/req = new()
+
+		req.prepare_roles_query(rank.group_id)
+		req.begin_async()
+
+		UNTIL(req.is_complete())
+
+		var/datum/http_response/resp = req.into_response()
+
+		if (resp.errored)
+			crash_with("Role request errored for id [rank.group_id] with: [resp.error]")
+			log_and_message_admins("Loading admins from forumuser API FAILED. Please alert web-service maintainers immediately!")
+			return
+
+		for (var/datum/forum_user/user in resp.body)
+			admins_to_push += user
+
+	// admin table is only cleared after ALL queries have been executed successfully.
+	clear_admins_table()
```

Comment:
An alternative to pre-clearing it could be something like that:
* Set a "status" flag in for each entry in the table
* Query the admins and write them to the table with `INSERT ON DUPLICATE KEY UPDATE` (and set the status flag to a value that indicates the entry has been created / updated
* If all inserts were successful, delete all entries in the table with the old status flag

The main difference with that is that this procedure prioritizes the presence of the admins where as truncate first prioritizes the absence of them (removal of admins that would have lost their perms in the last update)